store/tikv/coprocessor.go

```diff
@@ -239,7 +249,14 @@ func buildCopTasks(bo *Backoffer, cache *RegionCache, ranges *copRanges, req *kv
 				i = nextI
 			}
 		} else if req.StoreType == kv.TiFlash {
-			fullRange := kv.KeyRange{StartKey: regionWithRangeInfo.StartKey, EndKey: regionWithRangeInfo.EndKey}
+			left, right := regionWithRangeInfo.StartKey, regionWithRangeInfo.EndKey
+			if bytes.Compare(tableStart, left) >= 0 || len(left) == 0 {
```

Comment:
No need to check `len(left)` because if it is the case, previous check must be true.